commit 0c4758c69781808ee4293b746e3b16c77e304839
Author: Noralf Trønnes <noralf@tronnes.org>
Date:   Fri Sep 23 04:57:17 2016 +0200

    i2c: bcm2835: Support i2c-dev ioctl I2C_TIMEOUT
    
    Use i2c_adapter->timeout for the completion timeout value. The core
    default is 1 second.
    
    Signed-off-by: Noralf Trønnes <noralf@tronnes.org>
    Reviewed-by: Eric Anholt <eric@anholt.net>

diff --git a/drivers/i2c/busses/i2c-bcm2835.c b/drivers/i2c/busses/i2c-bcm2835.c
index 241e08ae7c27..d2085dd3742e 100644
--- a/drivers/i2c/busses/i2c-bcm2835.c
+++ b/drivers/i2c/busses/i2c-bcm2835.c
@@ -53,8 +53,6 @@
 #define BCM2835_I2C_CDIV_MIN	0x0002
 #define BCM2835_I2C_CDIV_MAX	0xFFFE
 
-#define BCM2835_I2C_TIMEOUT (msecs_to_jiffies(1000))
-
 struct bcm2835_i2c_dev {
 	struct device *dev;
 	void __iomem *regs;
@@ -242,7 +240,7 @@ static int bcm2835_i2c_xfer(struct i2c_adapter *adap, struct i2c_msg msgs[],
 	bcm2835_i2c_start_transfer(i2c_dev);
 
 	time_left = wait_for_completion_timeout(&i2c_dev->completion,
-						BCM2835_I2C_TIMEOUT);
+						adap->timeout);
 	if (!time_left) {
 		bcm2835_i2c_writel(i2c_dev, BCM2835_I2C_C,
 				   BCM2835_I2C_C_CLEAR);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1110_0c4758c69781808ee4293b746e3b16c77e304839.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1110_0c4758c69781808ee4293b746e3b16c77e304839.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
